pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -284,6 +286,58 @@ func (nm *NodeManager) DiscoverNode(nodeID string, searchBy cm.FindVM) error {
 
 	ipAddrNetworkNames := toIPAddrNetworkNames(nonVNICDevices)
 	nonLocalhostIPs := excludeLocalhostIPs(ipAddrNetworkNames)
+	guestInfo, encoding := extractGuestInfo(oVM.Config.ExtraConfig)
+
+	if guestInfo != "" && encoding == "base64" {
+		value, err := base64.StdEncoding.DecodeString(guestInfo)
+		if err != nil {
+			klog.Errorf("Error decoding guestinfo for vm=%+v in vc=%s and datacenter=%s: %v",
+				vmDI.VM, vmDI.VcServer, vmDI.DataCenter.Name(), err)
+			return err
+		}
+
+		type cloudInitConfig struct {
```

Comment:
Could we move this declaration out of the function?

Reply:
Done 👍